packages/adapter-flagsmith/src/index.ts

```diff
@@ -0,0 +1,54 @@
+import type { Adapter } from 'flags';
+import flagsmith, { IFlagsmithFeature, IInitConfig } from 'flagsmith';
+
+type AdapterParams = {
+  key?: string;
+} & IInitConfig;
+
+export function createFlagsmithAdapter<
+  ValueType extends IFlagsmithFeature,
+  EntitiesType,
+>({
+  key: customKey,
+  ...configParams
+}: AdapterParams): Adapter<ValueType, EntitiesType> {
+  async function initialize(config: IInitConfig) {
+    if (flagsmith?.initialised) return;
+    await flagsmith.init({ fetch: global.fetch, ...configParams });
+  }
+
+  return {
+    async decide({
+      key,
+      entities,
```

Comment:
Entities is unused right now, but it's where Flags SDK usually receives user information. I'd expect `identity` to be passed to `flagsmith.init({})` here if that's how your SDK works
https://docs.flagsmith.com/clients/next-ssr#comparing-ssr-and-client-side-flagsmith-usage

If you expect your customers to pass an identity, you should strongly consider using entities to drive that here.

What I'd expect is something like:

```ts
import type { Identity } from 'flagsmith'

// ...
ValueType extends IFlagsmithFeature,
EntitiesType extends IFlagsmithIdentity,

// ...
decide({ entities }) => {
  // identity is either entities itself, or a part of it
  init({ identity: entities })
}
```